src/openrct2/object/Object.cpp

```diff
@@ -329,6 +329,49 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+std::string VersionString(Version version)
+{
+    utf8 buff[VersionStringMaxLength];
+    return String::Format(
+        buff, VersionStringMaxLength, "%i.%i.%i", std::get<0>(version), std::get<1>(version), std::get<2>(version));
+}
+
+Version VersionTuple(std::string_view version)
+{
+    if (version.empty())
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[VersionNumFields] = {};
+    if (nums.size() > VersionNumFields)
+    {
+        log_warning("%i fields found in version string '%s', expected X.Y.Z", nums.size(), version);
+    }
+    if (nums.size() == 0)
+    {
+        log_warning("No fields found in version string '%s', expected X.Y.Z", version);
+        return std::make_tuple(0, 0, 0);
+    }
+    // fill in array from right-to-left
+    try
+    {
+        for (int8_t i = std::min(static_cast<int8_t>(nums.size()), VersionNumFields) - 1; i >= 0; i--)
+        {
+            uint8_t index = static_cast<uint8_t>(2 - (nums.size() - 1 + i));
+            uint16_t value = stoi(nums.at(i));
+            versions[index] = value;
```

Comment:
I got stack corruption errors in `versions` which I did not get before. I added the guard against indexing beyond the table and I cast the size_t to uint8_t which both together seemed to fix the error. Possibly what was causing problems before?